bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareDeviceHandler.java

```diff
@@ -0,0 +1,255 @@
+                                            newLedModeIdStr, device);
+                                }
+                            }
+                        }
+                    }
+                    break;
+                default:
+                    logger.warn("Update on unsupported channel {}, ignoring command", channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    protected void updateThing() {
+        SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+        if (device != null) {
+            logger.debug("Updating all thing channels for device : {}", device);
+            updateState(DEVICE_CHANNEL_ID, new DecimalType(device.id));
+            updateState(DEVICE_CHANNEL_NAME, new StringType(device.name));
+            updateState(DEVICE_CHANNEL_PRODUCT, new StringType(device.productId.toString()));
+            updateState(DEVICE_CHANNEL_ONLINE, OnOffType.from(device.status.online));
+
+            if (thing.getThingTypeUID().equals(THING_TYPE_HUB_DEVICE)) {
+                updateState(DEVICE_CHANNEL_LED_MODE, new StringType(device.status.ledModeId.toString()));
+                updateState(DEVICE_CHANNEL_PAIRING_MODE, new StringType(device.status.pairingModeId.toString()));
+            } else {
+                float batVol = device.status.battery;
+                updateState(DEVICE_CHANNEL_BATTERY_VOLTAGE, new DecimalType(batVol));
+                updateState(DEVICE_CHANNEL_BATTERY_LEVEL, new DecimalType(Math.min(
+                        (batVol - BATTERY_EMPTY_VOLTAGE) / (BATTERY_FULL_VOLTAGE - BATTERY_EMPTY_VOLTAGE) * 100.0f,
+                        100.0f)));
+                updateState(DEVICE_CHANNEL_LOW_BATTERY, OnOffType.from(batVol < LOW_BATTERY_THRESHOLD));
+                updateState(DEVICE_CHANNEL_DEVICE_RSSI, new DecimalType(device.status.signal.deviceRssi));
+                updateState(DEVICE_CHANNEL_HUB_RSSI, new DecimalType(device.status.signal.hubRssi));
+
+                if (thing.getThingTypeUID().equals(THING_TYPE_FLAP_DEVICE)) {
+                    updateThingCurfews(device);
+                    updateState(DEVICE_CHANNEL_LOCKING_MODE, new StringType(device.status.locking.modeId.toString()));
+                } else if (thing.getThingTypeUID().equals(THING_TYPE_FEEDER_DEVICE)) {
+                    int numBowls = device.control.bowls.bowlSettings.size();
+                    for (int i = 0; (i < 2) && (i < numBowls); i++) {
+                        BowlSettings bowlSettings = device.control.bowls.bowlSettings.get(i);
+                        if (device.control.bowls.bowlId.equals(1)) {
+                            updateState(DEVICE_CHANNEL_BOWLS_FOOD, new StringType(bowlSettings.foodId.toString()));
+                            updateState(DEVICE_CHANNEL_BOWLS_TARGET,
+                                    new QuantityType<Mass>(bowlSettings.targetId, SIUnits.GRAM));
+                        } else if (device.control.bowls.bowlId.equals(4)) {
```

Comment:
Why is this called inside of a for loop? If you have multiple settings these channels will have their values overwritten. Should each bowl have it's own channel?

Reply:
@HerzScheisse , could you comment on this one? As I don't have a feeder, I'm not sure what the API response is for the various settings.

Reply:
I dont know why i put this in a loop... its been a while and maybe i didnt find a better workaround for this...

Here is the JSON for these specific channels:

Do you guys know a better way to handle this?

`                "control": {
                    "bowls": {
                        "settings": [
                            {
                                "food_type": 2,
                                "target": 70
                            },
                            {
                                "food_type": 1,
                                "target": 100
                            }
                        ],
                        "type": 4
                    },
                    "fast_polling": false,
                    "lid": {
                        "close_delay": 4
                    },
                    "training_mode": 0
                },`

Reply:
Do you have a sample payload where the bowls type is `1`?

Reply:
Since the bowl types have some type of meaning can you perhaps make the known bowl ids constants in the code?

Reply:
> Do you have a sample payload where the bowls type is `1`?

Sure... here it is:

```
                "control": {
                    "bowls": {
                        "settings": [
                            {
                                "food_type": 2,
                                "target": 100
                            }
                        ],
                        "type": 1
                    },
                    "fast_polling": true,
                    "lid": {
                        "close_delay": 4
                    },
                    "training_mode": 0
                },
```

Reply:
I've changed this similar to the pet handler. The loop has been replaced with if statements based on bowl type.